dask_expr/_shuffle.py

```diff
@@ -244,7 +246,10 @@ def from_abstract_shuffle(cls, expr: Shuffle) -> Expr:
                 # Don't need to assign "_partitions" column
                 # if we are shuffling on a list of columns
                 nset = set(partitioning_index)
-                if nset & set(frame.columns) == nset and not index_shuffle:
+                if (
+                    nset & nset.intersection(set(frame.columns)) == nset
```

Comment:
This condition feels overly complicated. If I'm not missing anything, this simplifies to `nset.issubset(set(frame.columns))`.

Reply:
Removed altogether similar to dask/dask